operator/internal/probes/broker.go

```diff
@@ -0,0 +1,207 @@
+	//    controller id set.
+	//
+	// These allow us to ensure we aren't fully coupled to the overall cluster health status,
+	// which previously caused brokers to be marked as "not ready" when *any* broker in the cluster
+	// was unhealthy.
+
+	broker, err := client.Broker(ctx, brokerID)
+	if err != nil {
+		return false, fmt.Errorf("fetching broker status: %w", err)
+	}
+
+	// is the broker marked as active and not being decommissioned
+	if broker.MembershipStatus != rpadmin.MembershipStatusActive {
+		return false, nil
+	}
+
+	status, err := client.MaintenanceStatus(ctx)
+	if err != nil {
+		return false, fmt.Errorf("fetching broker maintenance status: %w", err)
+	}
+
+	// is the broker in maintenance mode and currently draining
+	if status.Draining {
+		return false, nil
+	}
+
+	summary, err := client.GetLocalPartitionsSummary(ctx)
+	if err != nil {
+		return false, fmt.Errorf("fetching broker partitions: %w", err)
+	}
+
+	// do we have any leaderless or under-replicated nodes?
+	if summary.Leaderless != 0 || summary.UnderReplicated != 0 {
+		return false, nil
+	}
+
+	clusterHealth, err := client.GetHealthOverview(ctx)
+	if err != nil {
+		return false, fmt.Errorf("fetching cluster health: %w", err)
+	}
+
+	// do we have an elected cluster leader (i.e. are we part of quorum)
+	if clusterHealth.ControllerID < 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// IsClusterBrokerReady checks if a cluster broker should be servicing requests. It does
+// this by issuing a request to the given broker which should get proxied through to the
+// underlying Kafka API, checking that the broker is marked as alive, not in maintenance
+// mode, and that we actually get a response should inform us that:
+//
+// 1. The Kafka API is up and servicing requests (through the broker active membership check).
+// 2. The Admin API is up and servicing requests (through receiving a valid request).
+// 3. We aren't currently draining in maintenance mode.
+func (p *Prober) IsClusterBrokerReady(ctx context.Context, brokerID int) (bool, error) {
+	client, err := p.getClient(ctx, brokerID)
+	if err != nil {
+		return false, fmt.Errorf("initializing client to check broker readiness: %w", err)
+	}
+	defer client.Close()
+
+	// unlike the above check which includes cluster quorum and partition status, this just
+	// checks that a broker is up and not in maintenance mode
+
+	broker, err := client.Broker(ctx, brokerID)
+	if err != nil {
+		return false, fmt.Errorf("fetching broker status: %w", err)
+	}
+
+	// is the broker marked as active and not being decommissioned
+	if broker.MembershipStatus != rpadmin.MembershipStatusActive {
+		return false, nil
+	}
+
+	status, err := client.MaintenanceStatus(ctx)
+	if err != nil {
+		return false, fmt.Errorf("fetching broker maintenance status: %w", err)
+	}
+
+	// is the broker in maintenance mode and currently draining
+	if status.Draining {
+		return false, nil
+	}
```

Comment:
I'm not sure if broker ready can relay only on one aspect of the maintenance status.

The response have many aspects like `finished`, `errors`, `failed` 
```go
type MaintenanceStatus struct {
	Draining     bool  `json:"draining"`
	Finished     *bool `json:"finished"`
	Errors       *bool `json:"errors"`
	Partitions   *int  `json:"partitions"`
	Eligible     *int  `json:"eligible"`
	Transferring *int  `json:"transferring"`
	Failed       *int  `json:"failed"`
}
```

For me ready would mean not in finished maintenance mode or not with maintenance that have errors.

In this situation I would ask core team to confirm.